src/Mollie/WC/Payment/Order.php

```diff
@@ -1100,4 +893,200 @@ protected function maximalFieldLengths($field, $maximalLength)
 
         return $field;
     }
+
+    /**
+     * @param WC_Order                    $order
+     * @param                             $new_order_status
+     * @param                             $order_id
+     * @param                             $paymentMethodTitle
+     * @param \Mollie\Api\Resources\Order $payment
+     */
+    protected function maybeUpdateStatus(
+        WC_Order $order,
+        $new_order_status,
+        $order_id,
+        $paymentMethodTitle,
+        \Mollie\Api\Resources\Order $payment
+    ) {
+        if (!$this->isOrderPaymentStartedByOtherGateway($order)) {
+            $gateway = wc_get_payment_gateway_by_order($order);
+
+            if ($gateway || ($gateway instanceof Mollie_WC_Gateway_Abstract)) {
```

Comment:
It looks like this `$gateway instanceof Mollie_WC_Gateway_Abstract` never makes sense :smiley:  If `$gateway` evaluated to `false`, so it cannot be an instance of this class. I noticed this before in the removed lines of code.
It shouldn't be harmful, though. 

But maybe there should be not `||` but `&&`? If so, this is a bug. This is the main reason why I'm writing this, actually.